packages/core/src/log.ts

```diff
@@ -0,0 +1,188 @@
+import type { Client } from './client';
+import { getClient, getCurrentScope } from './currentScopes';
+import { DEBUG_BUILD } from './debug-build';
+import type { Scope } from './scope';
+import { getDynamicSamplingContextFromScope } from './tracing';
+import type { DynamicSamplingContext, LogEnvelope, LogItem } from './types-hoist/envelope';
+import type { Log, LogAttribute, LogSeverityLevel } from './types-hoist/log';
+import { createEnvelope, dropUndefinedKeys, dsnToString, logger } from './utils-hoist';
+
+const LOG_BUFFER_MAX_LENGTH = 25;
+
+let GLOBAL_LOG_BUFFER: Log[] = [];
+
+let isFlushingLogs = false;
+
+const SEVERITY_TEXT_TO_SEVERITY_NUMBER: Partial<Record<LogSeverityLevel | 'log', number>> = {
+  trace: 1,
+  debug: 5,
+  info: 9,
+  log: 10,
+  warn: 13,
+  error: 17,
+  fatal: 21,
+};
+
+/**
+ * Creates envelope item for a single log
+ */
+export function createLogEnvelopeItem(log: Log): LogItem {
+  const headers: LogItem[0] = {
+    type: 'otel_log',
+  };
+
+  return [headers, log];
+}
+
+/**
+ * Records a log and sends it to sentry.
+ *
+ * Logs represent a message (and optionally some structured data) which provide context for a trace or error.
+ * Ex: sentry.addLog({level: 'warning', message: `user ${user} just bought ${item}`, attributes: {user, item}}
+ *
+ * @params log - the log object which will be sent
+ */
+function createLogEnvelope(logs: Log[], client: Client, scope: Scope): LogEnvelope {
+  const dsc = getDynamicSamplingContextFromScope(client, scope);
+
+  const dsn = client.getDsn();
+
+  const headers: LogEnvelope[0] = {
+    trace: dropUndefinedKeys(dsc) as DynamicSamplingContext,
+    ...(dsn ? { dsn: dsnToString(dsn) } : {}),
+  };
+
+  return createEnvelope<LogEnvelope>(headers, logs.map(createLogEnvelopeItem));
+}
+
+function valueToAttribute(key: string, value: unknown): LogAttribute {
+  switch (typeof value) {
+    case 'number':
+      return {
+        key,
+        value: { doubleValue: value },
+      };
+    case 'boolean':
+      return {
+        key,
+        value: { boolValue: value },
+      };
+    case 'string':
+      return {
+        key,
+        value: { stringValue: value },
+      };
+    default:
+      return {
+        key,
+        value: { stringValue: JSON.stringify(value) },
```

Comment:
```suggestion
        value: { stringValue: JSON.stringify(value) ?? "" },
```

`stringValue:` expects a value, not `undefined`